### PR TITLE
GDALDeserializeGCPListFromXML(), VRT and PAMDataset XMLInit(): make it take a const CPLXMLNode*

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: frmts/vrt/vrtderivedrasterband.cpp [hunk 1]
```diff
@@ -1457,10 +1457,11 @@ CPLErr VRTDerivedRasterBand::XMLInit(
         return CE_Failure;
     }
 
-    CPLXMLNode *psArgs = CPLGetXMLNode(psTree, "PixelFunctionArguments");
+    const CPLXMLNode *const psArgs =
+        CPLGetXMLNode(psTree, "PixelFunctionArguments");
     if (psArgs != nullptr)
     {
-        for (CPLXMLNode *psIter = psArgs->psChild; psIter != nullptr;
+        for (const CPLXMLNode *psIter = psArgs->psChild; psIter != nullptr;
```
Comment (elpaso) [inline frmts/vrt/vrtderivedrasterband.cpp:1464]:
```suggestion
        for (const CPLXMLNode *psIter = psArgs->psChild; psIter;
```

File: gcore/gdalpamdataset.cpp [hunk 2]
```diff
@@ -686,14 +682,12 @@ CPLErr GDALPamDataset::XMLInit(CPLXMLNode *psTree, const char *pszUnused)
...
         }
-        if (psValueAsXML)
-            CPLDestroyXMLNode(psValueAsXML);
     }
 
     /* -------------------------------------------------------------------- */
     /*      Process bands.                                                  */
     /* -------------------------------------------------------------------- */
-    for (CPLXMLNode *psBandTree = psTree->psChild; psBandTree != nullptr;
+    for (const CPLXMLNode *psBandTree = psTree->psChild; psBandTree != nullptr;
```
Comment (elpaso) [inline gcore/gdalpamdataset.cpp:690]:
```suggestion
    for (const CPLXMLNode *psBandTree = psTree->psChild; psBandTree;
```

File: gcore/gdalpamrasterband.cpp [hunk 3]
```diff
@@ -450,13 +448,12 @@ CPLErr GDALPamRasterBand::XMLInit(CPLXMLNode *psTree,
...
-    const auto psCategoryNames = CPLGetXMLNode(psTree, "CategoryNames");
-    if (psCategoryNames)
+    if (const auto psCategoryNames = CPLGetXMLNode(psTree, "CategoryNames"))
     {
         CPLStringList oCategoryNames;
 
-        for (CPLXMLNode *psEntry = psCategoryNames->psChild; psEntry != nullptr;
-             psEntry = psEntry->psNext)
+        for (const CPLXMLNode *psEntry = psCategoryNames->psChild;
+             psEntry != nullptr; psEntry = psEntry->psNext)
```
Comment (elpaso) [inline gcore/gdalpamrasterband.cpp:456]:
```suggestion
             psEntry; psEntry = psEntry->psNext)
```

File: gcore/gdalpamrasterband.cpp [hunk 4]
```diff
@@ -475,14 +472,13 @@ CPLErr GDALPamRasterBand::XMLInit(CPLXMLNode *psTree,
...
-    if (psColorTable)
+    if (const auto psColorTable = CPLGetXMLNode(psTree, "ColorTable"))
     {
         GDALColorTable oTable;
         int iEntry = 0;
 
-        for (CPLXMLNode *psEntry = psColorTable->psChild; psEntry != nullptr;
-             psEntry = psEntry->psNext)
+        for (const CPLXMLNode *psEntry = psColorTable->psChild;
+             psEntry != nullptr; psEntry = psEntry->psNext)
```
Comment (elpaso) [inline gcore/gdalpamrasterband.cpp:481]:
```suggestion
             psEntry; psEntry = psEntry->psNext)
```

**Review by elpaso — APPROVED** [review]

File: frmts/vrt/vrtderivedrasterband.cpp (hunk 1 above)
Comment (rouault) [reply frmts/vrt/vrtderivedrasterband.cpp]: done

File: gcore/gdalpamdataset.cpp (hunk 2 above)
Comment (rouault) [reply gcore/gdalpamdataset.cpp]: done

File: gcore/gdalpamrasterband.cpp (hunk 3 above)
Comment (rouault) [reply gcore/gdalpamrasterband.cpp]: done

File: gcore/gdalpamrasterband.cpp (hunk 4 above)
Comment (rouault) [reply gcore/gdalpamrasterband.cpp]: done

**Pull request merged**